### ISSUE TITLE
async-io dependency breaks build

### ISSUE THREAD

**Issue by FrozenDroid** [ivmarkov/edge-net#7]
Quite easy to reproduce - just build master and you will see many errors involving libc.

**Comment by FrozenDroid** [comment]
```
[patch.crates-io]
async-io = { git = "https://github.com/smol-rs/async-io.git", rev = "d9c9ed8" }
socket2 = { git = "https://github.com/rust-lang/socket2", rev = "77e3bb9" }
```
seems to fix my issue - I suppose we need to wait for async-io and socket2 to release new versions to fix this.

**Comment by ivmarkov** [comment]
Upstream support for ESP IDF in the `smol` ecosystem is in an odd shape right now:

We need new releases for the following crates, as the ESP IDF changes are upstreamed in there, but not released yet:
* `socket2`
* `polling`
* `rust-errno`

Crates where the ESP IDF changes are upstreamed *and also* released:
* `rustix`

Crates where ESP IDF changes are upstreamed, yet their `master` branch is not usable yet:
* `async-io`, because of [this](https://github.com/smol-rs/async-io/pull/142)

In the meantime, what you can do is:
* pin `async-io` to an old version as done [here](https://github.com/ivmarkov/rust-esp32-std-demo/blob/main/Cargo.toml#L78) and [here](https://github.com/project-chip/matter-rs/blob/no_std/matter/Cargo.toml#L54)
* Patch `polling` and `socket2` (and possibly, the `smol` aggregation crate itself) as done [here](https://github.com/ivmarkov/rust-esp32-std-demo/blob/main/Cargo.toml#L14) to earlier not upstreamed forks. also done [here](https://github.com/project-chip/matter-rs/blob/no_std/Cargo.toml#L7)

**Comment by ivmarkov** [comment]
@FrozenDroid Small progress update: 
* `polling` 3.0 was released recently, with all ESP-IDF-related changes upstreamed
* The relevant ESP-IDF changes to `rustix`,`socket2` and `errno` are also upstreamed and released. 

Waiting for the release of `async-io` 2.0 now, which [I hope is imminent](https://github.com/smol-rs/async-io/pull/155), and which would probably be released together with `smol` 2.0.

That would complete the upstreaming story of ESP-IDF in the `smol-rs` echosystem.

**Comment by ivmarkov** [comment]
https://github.com/ivmarkov/edge-net/commit/d96b37b724ee45556df092287015b5cf7ee09735
